test/functional/measures_controller_measure_history_granular_test.rb

```
@@ -0,0 +1,215 @@
+    assert_nil upload_summary.first.measure_db_id_pre_upload
+    assert_equal upload_summary.first.measure_db_id_post_upload, measure.id
+    assert_nil upload_summary.first.cms_id_pre_upload
+    assert_equal upload_summary.first.cms_id_post_upload, measure.cms_id
+    assert_nil upload_summary.first.hqmf_version_number_pre_upload
+    assert_equal upload_summary.first.hqmf_version_number_post_upload, measure.hqmf_version_number
+    assert_equal 1, upload_summary.first.population_set_summaries.count 
+    assert_equal 0, upload_summary.first.measure_population_set_count[:pre_upload]
+    assert_equal 1, upload_summary.first.measure_population_set_count[:post_upload]
+    assert_equal 0, upload_summary.first.population_set_summaries[0][:patients].count
+    assert_equal 2, upload_summary.first.population_set_summaries[0][:summary].count
+    assert_equal 0, upload_summary.first.population_set_summaries[0][:summary][:pass_after]
+    assert_equal 0, upload_summary.first.population_set_summaries[0][:summary][:fail_after]
+    
+    # Load the patients for the measure
+    records_set = File.join('records','measure_history_set', 'base_example')
+    collection_fixtures(records_set)
+    @patients = Record.where(measure_ids: '9A032D9C-3D9B-11E1-8634-00237D5BF174')
+    associate_user_with_patients(@user, @patients)
+    @patients.each do |patient|
+      assert_equal measure.hqmf_set_id, patient[:measure_ids][0]
+      patient['user_id'] = BSON::ObjectId.from_string(patient['user_id']) if patient['user_id'].is_a?(String)
+      patient.unset(:calc_results)
+      patient.save!
+    end
+
+    calculator = BonnieBackendCalculator.new
+    patients = Record.where(user_id: measure.user_id, measure_ids: measure.hqmf_set_id)
+    # Move on if there are no patients associated with this measure for this user
+    if patients.count == 0
+      next
+    end
+    measure.populations.each_with_index do |population_set, population_index|
+      begin
```

we don't want a rescue here. we want to explicitly fail in our test if something unexpected happens.